### PR TITLE
Add report + moderation messaging

### PR DESCRIPTION
**Pull request by Geometrically**
Fixes MOD-117
Fixes MOD-156
Fixes MOD-157

### CODE REVIEW

File: src/models/threads.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    pub fn from_str(string: &str) -> ThreadType {
+        match string {
+            "required" => ThreadType::Report,
+            "optional" => ThreadType::Project,
+            "unsupported" => ThreadType::DirectMessage,
+            _ => ThreadType::DirectMessage,
+        }
```
Comment (triphora) [inline src/models/threads.rs:73]: These need to be updated

File: src/routes/v2/mod.rs [hunk 2]
```diff
@@ -32,6 +33,7 @@ pub fn config(cfg: &mut actix_web::web::ServiceConfig) {
             .configure(teams::config)
             .configure(users::config)
             .configure(version_file::config)
-            .configure(versions::config),
+            .configure(versions::config)
+            .configure(threads::config),
```
Comment (triphora) [inline src/routes/v2/mod.rs:37]: Keep the alphabetical order :irritatered:

File: src/routes/v2/threads.rs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    }
+    Ok(HttpResponse::NotFound().body(""))
+}
+
+#[derive(Deserialize)]
+pub struct NewThreadMessage {
+    pub body: MessageBody,
+}
+
+#[patch("{id}")]
```
Comment (triphora) [inline src/routes/v2/threads.rs:126]: Seems like this should be a POST route

**Review by triphora — DISMISSED** [review]
Is there be a way for moderators to see existing *project* threads (not report threads) that are unresolved? Also, how does opening/closing project threads work?

**Review by Prospector — CHANGES_REQUESTED** [review]
Thread messages need a way to differentiate anonymous moderator from deleted user or a user that is null for potentially other reasons.

Also, it'd be nice if the project status change message type included the old project status in the message

**Review by thesuzerain — APPROVED** [review]

**Pull request merged**